File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/TaskSelection.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.components.toolbar.SelectionOperation;
import io.intino.alexandria.ui.displays.events.SelectionEvent;
import io.intino.alexandria.ui.displays.events.SelectionListener;
import io.intino.alexandria.ui.displays.notifiers.TaskSelectionNotifier;

import java.util.Collections;
import java.util.List;

public abstract class TaskSelection<DN extends TaskSelectionNotifier, B extends Box> extends AbstractTaskSelection<DN, B> implements SelectionOperation {

    public TaskSelection(B box) {
        super(box);
    }

}

File: framework/ui-elements/src/displays/components/TaskSelection.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractTaskSelection from "../../../gen/displays/components/AbstractTaskSelection";
import TaskSelectionNotifier from "../../../gen/displays/notifiers/TaskSelectionNotifier";
import TaskSelectionRequester from "../../../gen/displays/requesters/TaskSelectionRequester";
import Operation from "./Operation"
import { withSnackbar } from 'notistack';
import DisplayFactory from "alexandria-ui-elements/src/displays/DisplayFactory";

class TaskSelection extends AbstractTaskSelection {
	constructor(props) {
		super(props);
		this.notifier = new TaskSelectionNotifier(this);
		this.requester = new TaskSelectionRequester(this);
	};
}

export default withStyles(Operation.Styles, { withTheme: true })(withSnackbar(TaskSelection));
DisplayFactory.register("TaskSelection", withStyles(Operation.Styles, { withTheme: true })(withSnackbar(TaskSelection)));